pennylane/labs/resource_estimation/templates/qubitize.py

```diff
@@ -30,14 +30,23 @@
 
 
 class ResourceQubitizeTHC(ResourceOperator):
-    r"""Resource class for Qubitization of THC Hamiltonian.
+    r"""Resource class for qubitization of a Hamiltonian.
+
+    .. note::
+
+            This decomposition assumes that an appropriately sized phase gradient state is available.
+            Users should ensure that the cost of constructing this state has been accounted for.
+            See also :class:`~.pennylane.labs.resource_estimation.ResourcePhaseGradient`.
 
     Args:
         compact_ham (~pennylane.labs.resource_estimation.CompactHamiltonian): a tensor hypercontracted
             Hamiltonian for which the walk operator is being created
-        coeff_precision (float, optional): precision for loading the coefficients of Hamiltonian
-        rotation_precision (float, optional): precision for loading the rotation angles for basis rotation
-        compare_precision (float, optional): precision for comparing two numbers
+        prep_op (Union[~pennylane.labs.resource_estimation.ResourceOperator, None]): An optional
+            resource operator, corresponding to the prepare routine. If :code:`None`, the
+            default :class:`~.pennylane.labs.resource_estimation.ResourcePrepTHC` will be used.
+        select_op (Union[~pennylane.labs.resource_estimation.ResourceOperator, None]): An optional
+            resource operator, corresponding to the select routine. If :code:`None`, the
+            default :class:`~.pennylane.labs.resource_estimation.ResourceSelectTHC` will be used.
         wires (list[int] or optional): the wires on which the operator acts
 
     Resources:
```

Comment:
Can we add an advanced example here as well where a user prepares modified instances of PrepTHC and SelectTHC (maybe changing the SelSwapDepth or the coefficient precisions) and passes them into this class along with the hamiltonian.